nbsphinx_link/__init__.py

```diff
@@ -59,6 +131,11 @@ def parse(self, inputstring, document):
         path = utils.relative_path(None, abs_path)
         path = nodes.reprunicode(path)
 
+        extra_media = link.get('extra-media', None)
+        if extra_media:
+            source_file = env.doc2path(env.docname)
+            collect_extra_media(extra_media, source_file, path)
```

Comment:
I think we need to mark all our input files as a dependency (see below). This will help ensure that Sphinx re-runs this parser if some of those files change (rather than skipping this build step since the linkfile + notebook are unchanged).

Reply:
I implemented it, but I'm not sure if it works properly directorys.
I tryed to trace back the definitions ``note_dependency`` and ``ecord_dependencies.add``, but couldn't find them.

Reply:
I think we can use [`note_reread`](https://www.sphinx-doc.org/en/master/extdev/envapi.html#sphinx.environment.BuildEnvironment.note_reread):
- If `extra-media` includes a directory, call `note_reread` to make sure the current file is always rebuilt. Each individual file should still be added to `record_dependencies` (and probably also good to add with `note_dependency`).
- If `extra-media` is not present, or only has files, add all the files with `note_dependency` etc.

Reply:
Wouldn't ``note_reread`` rerender the notebook no matter if it changed or not?

Reply:
> Wouldn't `note_reread` rerender the notebook no matter if it changed or not?

Yes, but that is the only way to ensure that all changes in a media directory gets included. Take this case:
1. You add a reference to an image 'images/bob.png' in a markdown cell, add the dir to the media field, and then make the docs, which see the changes to the notebook and rebuilds. ✔️ 
2. You want to replace the image with another, so you delete 'bob.png', but forget to copy over the new image (or the copy op fails for some reason). You make the docs, which see the changes to bob.png (deletion) and rebuilds. ✔️
3. You see the broken image in your docs, and realize your mistake. You add the new image and make the docs. No previously used file (link, notebook, or image file) has changed, so it skips the build. ❌

Reply:
(Actually, the above example might fail at step 2 due to a missing reference, and failed build will always rebuild, but the concept holds for other examples)

Reply:
Now each file is registered indiviualy with ``note_dependency`` and ``record_dependencies``

Reply:
But no `note_reread`?

Reply:
Maybe I missunderstood your comment:

> If extra-media is not present, or only has files, add all the files with note_dependency etc.

Reply:
Do you want me add ``note_reread`` on a per file basis, for the extra media files?
I would add a ``is_extra_media`` kwarg to ``register_dependency``.

Reply:
I meant: If `extra-media` is present, and contains at least one directory, then run `document.settings.env.note_reread()` once. The other code should run in all cases (might not be needed, but let's be defensive).

Reply:
Now it runs once if ``extra-media`` is present at all.
Else we would need an extra loop/map, after copying the files, to check if any item of ``extra_media`` is a directory, so it won't run on each directory.

Reply:
I pushed a commit that checks each config arg as part of the outer loop, and call `note_reread` the first time a dir is found.

Reply:
Let me know if you agree with the change.